### PR TITLE
feat: Write tests for kyc operations

### PR DESCRIPTION
**Pull request by stefan-stefanooov**
**Description**:
- grantTokenKyc
- revokeTokenKyc

Suite | Tests | Signatures
-- | -- | --
GrantRevokeKycV1SecurityModelSuite | grantRevokeKycSpec | - transaction payer- contract Id- KYC key

**Related issue(s)**:
https://github.com/hashgraph/hedera-services/issues/12725
Fixes #
https://github.com/hashgraph/hedera-services/issues/12725

### CODE REVIEW

**Review by petreze — COMMENTED** [review]
The changes should be in `GrantRevokeKycV1SecurityModelSuite` to enable and fix `grantRevokeKycSpec()` test

**Comment by stefan-stefanooov** [comment]
> The changes should be in `GrantRevokeKycV1SecurityModelSuite` to enable and fix `grantRevokeKycSpec()` test

I think the new test should be in `GrantRevokeKycSuite` file. 
The file name with the `v1` tag in it, holds the v1 security model tests. The one no tag, holds the v2 (current) security model tests.
This pattern is held in all tests
image

Do you agree ?

**Review by petreze — COMMENTED** [review]
aah yes, I see. Didn't notice that pattern, good point!

**Review by petreze — DISMISSED** [review]
LGTM, ty

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/GrantRevokeKycSuite.java [hunk 1]
```diff
@@ -340,4 +338,105 @@ final HapiSpec grantRevokeKycSpecWithAliasLocalCall() {
...
+    @HapiTest
+    final HapiSpec grantRevokeKycSpec() {
+        final AtomicReference<TokenID> vanillaTokenID = new AtomicReference<>();
+        final AtomicReference<AccountID> accountID = new AtomicReference<>();
+        final AtomicReference<AccountID> secondAccountID = new AtomicReference<>();
+
+        return propertyPreservingHapiSpec("grantRevokeKycSpec")
+                .preserving(CONTRACTS_MAX_NUM_WITH_HAPI_SIGS_ACCESS)
+                .given(
+                        overriding(CONTRACTS_MAX_NUM_WITH_HAPI_SIGS_ACCESS, CONTRACTS_V1_SECURITY_MODEL_BLOCK_CUTOFF),
```
[the remark this answers is not in the document]
Comment (stefan-stefanooov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/GrantRevokeKycSuite.java]: Removed

**Review by petreze — DISMISSED** [review]
LGTM

**Review by MrValioBg — APPROVED** [review]
LGTM!

**Review by petreze — DISMISSED** [review]
LGTM

File: hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/GrantRevokeKycSuite.java [hunk 2]
```diff
@@ -340,4 +354,103 @@ final HapiSpec grantRevokeKycSpecWithAliasLocalCall() {
...
+                        uploadInitCode(GRANT_REVOKE_KYC_CONTRACT),
+                        contractCreate(GRANT_REVOKE_KYC_CONTRACT),
+                        tokenAssociate(ACCOUNT, VANILLA_TOKEN),
+                        tokenAssociate(SECOND_ACCOUNT, VANILLA_TOKEN),
+                        newKeyNamed(THRESHOLD_KEY)
+                                .shape(THRESHOLD_KEY_SHAPE.signedWith(sigs(ON, GRANT_REVOKE_KYC_CONTRACT))),
+                        tokenUpdate(VANILLA_TOKEN).kycKey(THRESHOLD_KEY).signedByPayerAnd(ADMIN_KEY))
+                .when(withOpContext((spec, opLog) -> allRunFor(
+                        spec,
+                        contractCall(
```
[the remark this answers is not in the document]
Comment (stefan-stefanooov) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/GrantRevokeKycSuite.java:390]: @lukelee-sl  Hey ! 
There is such a test in the Suite. It's called `grantRevokeKycFail` written by someone before me.

[the remark this answers is not in the document]
Comment (lukelee-sl) [reply hedera-node/test-clients/src/main/java/com/hedera/services/bdd/suites/contract/precompile/GrantRevokeKycSuite.java:538]: Gotcha.  Thanks for checking.

**Review by lukelee-sl — APPROVED** [review]
LGTM

**Pull request merged**